client/executor/wasmtime/src/runtime.rs

```diff
@@ -412,25 +455,32 @@ unsafe fn do_create_runtime(
 	})
 }
 
-fn instrument(blob: &mut RuntimeBlob, semantics: &Semantics) {
+fn instrument(
+	mut blob: RuntimeBlob,
+	semantics: &Semantics,
+) -> std::result::Result<RuntimeBlob, WasmError> {
+	if semantics.stack_depth_metering {
+		const STACK_DEPTH_LIMIT: u32 = 65536;
```

Comment:
the reason for hardcoding it is that we don't accidentally have a consensus issue?
so it's expected to never change (at least never go down)?

Reply:
I wouldn't say that this is the reason. I am not sure if this solution provides a lot, in terms of accidentally having different numbers. I am not sure if it will stay the same forever either. It's just its simpler this way.

I would imagine we would still be able to change this number by first making a change that sources this number from, e.g. persisted validation data / host configuration and then when sufficient number of nodes are updated we can change this through the configuration.

If the number is different on different machines and somebody probes how deep recursion can be, this can lead to a split and consequently to disputes.